resources/packs/k8s/k8s.lr

```diff
@@ -540,6 +542,32 @@ k8s.service @defaults("namespace name created") {
   spec dict
 }
 
+// Kubernetes Ingress
+k8s.ingress @defaults("namespace name created") {
+  // Mondoo ID for Kubernetes Object
+  id string
+  // Kubernetes Object UID
+  uid string
+  // Kubernetes Resource Version
+  resourceVersion string
+  // Kubernetes Labels
+  labels() map[string]string
+  // Kubernetes Annotations
+  annotations() map[string]string
+  // Kubernetes Object Name
+  name string
+  // Kubernetes Object Namesapce
+  namespace string
+  // Kubernetes Object Type
+  kind string
+  // Kubernetes Object Creation Timestamp
+  created time
+  // Full resource manifest
+  manifest dict
+  // Ingress Spec
+  spec dict
```

Comment:
don't we want a specific type for the ingress spec? Or at least take out the ingress props in the ingress object? The main thing we want to check is in the spec and right now it's just a dict. I guess if we want to take out those props we should probably do it already in part 1 since otherwise we will need to re-write any policies that have been written in-between.

Reply:
It isn't clear to me right now which fields should be elevated outside of the .spec fields. There is definite upcoming work to create the new 'certificates' field that will be generated based out of the Secrets referenced in .spec.tls[], but that is in a separate PR #607

Reply:
I think we can have strong types for path blocks. Now if I want to query my endpoints I need to use JSON and figure my way through it. Instead we can have an actual type that lets us query the endpoints we want and have autocompletion